tests/integration/test_convolve.py

```diff
@@ -13,15 +13,66 @@
 # limitations under the License.
 #
 
+import os
+
 import numpy as np
 import pytest
 import scipy.signal as sig
 from utils.comparisons import allclose
 
 import cunumeric as num
 
-SHAPES = [(100,), (10, 10), (10, 10, 10)]
-FILTER_SHAPES = [(5,), (3, 5), (3, 5, 3)]
+CUDA_TEST = os.environ.get("LEGATE_NEED_CUDA") == "1"
+
+SHAPES = [(100,), (10, 10), (10, 10, 10), (32, 2, 32)]
+FILTER_SHAPES = [(5,), (3, 5), (3, 5, 3), (32, 1, 32)]
+
+LARGE_SHAPES = [
+    pytest.param(
+        (128, 2, 1024),
+        (64, 2, 512),
+        marks=pytest.mark.xfail(
+            not CUDA_TEST, run=False, reason="test hang on CPU variants"
```

Comment:
It doesn't really make sense to me to supply a `condition` value to `xfail` and also `run=False` at the same time. Did you mean to set `run` based on `CUDA_TEST` as well? If we are never going to run them in any case, let's just mark them `skip` and dispense with the env var altogether.

Reply:
> Did you mean to set run based on CUDA_TEST as well? 

Yes, the idea is to skip the params on cpu variants for now but only because they block everything else, and they really should be fixed, so marking them as `xfail` so we know it's something that needs to be fixed and we can check the fix or debug later with `--runxfail`. Test will be run normally on GPU variants.

E.g. this param on GPU:
```
tests/integration/test_convolve.py::test_double[(128, 2, 1024)-(64, 2, 512)] PASSED [ 19%] 
```

on CPU:
```
tests/integration/test_convolve.py::test_double[(128, 2, 1024)-(64, 2, 512)] XFAIL [ 19%] 
```

Let me know if you still prefer dispensing the params with the env var and then modify the condition when a fix is available, and I'll update the script accordingly.

Reply:
>  Test will be run normally on GPU variants.

Doesn't unconditional `run=False` mean that the test will never run in *any* case? Maybe I am misunderstanding something, but I would have expected that `run=CUDA_TEST` would be what is needed in order for:
* the test to run normally in the GPU case
* the test to not run at all (and also be `xfailed`) in the CPU case

Are you seeing the test actually run in a GPU configuration with `run=False`?

Reply:
When there's a condition specified for `xfail` , `run=False` is also conditional since if the condition is not met the whole `xfail` is just ignored.

Just as a demonstration, I added a dummy message to the test body so the test looks like this:
```
@pytest.mark.parametrize(
    "shape, filter_shape",
    list(zip(SHAPES, FILTER_SHAPES)) + LARGE_SHAPES,
    ids=str,
)
def test_double(shape, filter_shape):
    print(f'running with {shape} and {filter_shape}')  <<<<<<<<<<<<<<<<
    a = num.random.rand(*shape)
    v = num.random.rand(*filter_shape)

    check_convolve(a, v)
    check_convolve(v, a)
    print(f'finished running with {shape} and {filter_shape}')       <<<<<<<<<<<<<<<<<<<
```

On GPU:

```
(legate) root@c7f115c3207c:/opt/legate/cunumeric# legate --gpus 1 --fbmem 8192 tests/integration/test_convolve.py --capture=no --verbose -k test_double
======================================================================= test session starts =======================================================================
platform linux -- Python 3.9.18, pytest-7.4.3, pluggy-1.3.0 -- /opt/conda/envs/legate/bin/python3
cachedir: .pytest_cache
rootdir: /opt/legate/cunumeric
configfile: pyproject.toml
plugins: lazy-fixture-0.6.3, mock-3.12.0, cov-4.1.0
collected 41 items / 33 deselected / 8 selected

tests/integration/test_convolve.py::test_double[(100,)-(5,)] running with (100,) and (5,)
finished running with (100,) and (5,)
PASSED
tests/integration/test_convolve.py::test_double[(10, 10)-(3, 5)] running with (10, 10) and (3, 5)
finished running with (10, 10) and (3, 5)
PASSED
tests/integration/test_convolve.py::test_double[(10, 10, 10)-(3, 5, 3)] running with (10, 10, 10) and (3, 5, 3)
finished running with (10, 10, 10) and (3, 5, 3)
PASSED
tests/integration/test_convolve.py::test_double[(32, 2, 32)-(32, 1, 32)] running with (32, 2, 32) and (32, 1, 32)
finished running with (32, 2, 32) and (32, 1, 32)
PASSED
tests/integration/test_convolve.py::test_double[(128, 2, 1024)-(64, 2, 512)] running with (128, 2, 1024) and (64, 2, 512)
finished running with (128, 2, 1024) and (64, 2, 512)            <<<<<<<<<<<<<<<<<<<<<<<<<<<<
PASSED
tests/integration/test_convolve.py::test_double[(1024, 2, 8192)-(128, 16, 64)] running with (1024, 2, 8192) and (128, 16, 64)
finished running with (1024, 2, 8192) and (128, 16, 64)
PASSED
tests/integration/test_convolve.py::test_double[(1024, 2, 1024)-(5, 1, 5)] XFAIL ([NOTRUN] GPU variant hits SIGABRT)
tests/integration/test_convolve.py::test_double[(1024, 2, 1024)-(128, 1, 128)] XFAIL ([NOTRUN] GPU variant hits SIGFPE, CPU hangs)

========================================================== 6 passed, 33 deselected, 2 xfailed in 32.02s ===========================================================

```

On CPU:
```
(legate) root@c7f115c3207c:/opt/legate/cunumeric# legate --cpus 1 tests/integration/test_convolve.py --capture=no --verbose -k test_double
======================================================================= test session starts =======================================================================
platform linux -- Python 3.9.18, pytest-7.4.3, pluggy-1.3.0 -- /opt/conda/envs/legate/bin/python3
cachedir: .pytest_cache
rootdir: /opt/legate/cunumeric
configfile: pyproject.toml
plugins: lazy-fixture-0.6.3, mock-3.12.0, cov-4.1.0
collected 41 items / 33 deselected / 8 selected

tests/integration/test_convolve.py::test_double[(100,)-(5,)] running with (100,) and (5,)
finished running with (100,) and (5,)
PASSED
tests/integration/test_convolve.py::test_double[(10, 10)-(3, 5)] running with (10, 10) and (3, 5)
finished running with (10, 10) and (3, 5)
PASSED
tests/integration/test_convolve.py::test_double[(10, 10, 10)-(3, 5, 3)] running with (10, 10, 10) and (3, 5, 3)
finished running with (10, 10, 10) and (3, 5, 3)
PASSED
tests/integration/test_convolve.py::test_double[(32, 2, 32)-(32, 1, 32)] running with (32, 2, 32) and (32, 1, 32)
finished running with (32, 2, 32) and (32, 1, 32)
PASSED
tests/integration/test_convolve.py::test_double[(128, 2, 1024)-(64, 2, 512)] XFAIL ([NOTRUN] test hang on CPU variants)
tests/integration/test_convolve.py::test_double[(1024, 2, 8192)-(128, 16, 64)] XFAIL ([NOTRUN] test hang on CPU variants)
tests/integration/test_convolve.py::test_double[(1024, 2, 1024)-(5, 1, 5)] running with (1024, 2, 1024) and (5, 1, 5)
finished running with (1024, 2, 1024) and (5, 1, 5)
PASSED
tests/integration/test_convolve.py::test_double[(1024, 2, 1024)-(128, 1, 128)] XFAIL ([NOTRUN] GPU variant hits SIGFPE, CPU hangs)

=========================================================== 5 passed, 33 deselected, 3 xfailed in 5.04s ===========================================================
```

As you can see the tests are still running on the expected configuration, and if they're not run pytest tells you it's not run (though not displayed when running with test.py)